### PR TITLE
Prepare package for public consumption and contributions

### PR DESCRIPTION
**Pull request by cottsay**
For the most part, these changes are superficial and linter-related.

The most notable exception is that I corrected the namespace delimiter in the parameters from a `/` to a `.`, which will need a corresponding update in the OpenRover config files.

Once the `serial` package is available in ROS 2, we can think about submitting this as a standard ROS 2 package, at which point we can leverage the ROS 2 buildfarm for CI and PR builds of this repository.

These updates make the package buildable on OSX and Windows as well, though the end-to-end smoke test only runs on UNIX-based systems, so it is excluded from Windows testing.

After this is merged, I'd like to tag the first release of the package.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+      License. However, in accepting such obligations, You may act only
+      on Your own behalf and on Your sole responsibility, not on behalf
+      of any other Contributor, and only if You agree to indemnify,
+      defend, and hold each Contributor harmless for any liability
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
+   APPENDIX: How to apply the Apache License to your work.
```
Comment (rotu) [inline LICENSE:179]: You shouldn't need the whole license file, and you don't identify the copyright holder in the license file. You can delete almost this entire file and fill out the template listed here, in the appendix.

File: src/bno055_driver.cpp [hunk 2]
```diff
@@ -181,39 +181,42 @@ rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
...
+    port_->write_command_.data[0] = magnetometer_radius & 0xFF;
+    port_->write_command_.data[1] = (magnetometer_radius >> 8) & 0xFF;
+    port_->write();
   }
 
   imu_msg_ = std::make_shared<sensor_msgs::msg::Imu>();
   imu_pub_ = this->create_publisher<sensor_msgs::msg::Imu>("imu/data", rclcpp::SensorDataQoS());
   mag_msg_ = std::make_shared<sensor_msgs::msg::MagneticField>();
-  mag_pub_ = this->create_publisher<sensor_msgs::msg::MagneticField>("imu/mag", rclcpp::SensorDataQoS());
+  mag_pub_ = this->create_publisher<sensor_msgs::msg::MagneticField>("imu/mag",
```
Comment (rotu) [inline src/bno055_driver.cpp]: This violates the [ROS2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#open-versus-cuddled-braces), but it seems to be a defect in `ament_uncrustify`, so I'm okay leaving it as-is for now.

Submitted a PR: https://github.com/ament/ament_lint/pull/148

File: src/bno055_driver.cpp [hunk 3]
```diff
@@ -562,7 +566,7 @@ try
 
   diag_pub_->publish(*diag_msg_);
 } catch (std::exception & e) {
-  RCLCPP_ERROR(get_logger(), "Failed to poll and publish diagnostics");
+  RCLCPP_ERROR(get_logger(), "Failed to poll and publish diagnostics: %s", e.what());
```
Comment (rotu) [inline src/bno055_driver.cpp:579]: :+1:

File: src/bno055_uart.cpp [hunk 4]
```diff
@@ -59,57 +60,60 @@ bool BNO055UART::isOpen()
 void BNO055UART::read(uint8_t page)
 {
   if (page != page_ && read_command_.address != BNO055Register::PAGE_ID) {
-    // TODO: Change pages automtically
+    // TODO(cottsay): Change pages automtically
```
Comment (rotu) [inline src/bno055_uart.cpp]: Typo. Automatically or atomically?

File: src/bno055_uart.cpp [hunk 5]
```diff
@@ -59,57 +60,60 @@ bool BNO055UART::isOpen()
...
     throw serial::SerialException("Page change required");
   }
 
   for (size_t this_write, index = 0; index < sizeof(read_command_); index += this_write) {
-    this_write = port_.write((const uint8_t *)&read_command_ + index, 1);
+    this_write = port_.write(reinterpret_cast<const uint8_t *>(&read_command_) + index, 1);
```
Comment (rotu) [inline src/bno055_uart.cpp]: I'm getting a major code smell here, and it's hard to tell what you're doing. If you must do it this way,
1. move the reinterpret cast outside the loop.
2. get rid of `this_write`. If it's zero, you're throwing an exception and you're passing 1 for the length.

Better yet, use port_.write(reinterpret_cast<const uint8_t *>(&read_command_),sizeof(read_command_))

**Review by rotu — CHANGES_REQUESTED** [review]
Looks pretty good. Only some minor changes

File: src/bno055_driver.cpp (hunk 2 above)
Comment (cottsay) [reply src/bno055_driver.cpp]: I'll update the package to be compliant regardless of the `nl_func_call_start_multi_line` value.

File: LICENSE (hunk 1 above)
Comment (cottsay) [reply LICENSE:179]: This particular license may not require distribution with the product, but some do. For that reason, many operating systems require that the license always be distributed with sources and/or binary products.

Furthermore, a `LICENSE` and `CONTRIBUTING.md` are both required to pass the `ament_copyright` tests.

I believe that the APPENDIX in this license file is intended to be a template. If someone were to contribute a new file, they'd fill in their own copyright information, so I don't believe there is any value in filling it in here.

File: src/bno055_uart.cpp (hunk 5 above)
Comment (cottsay) [reply src/bno055_uart.cpp]: I refactored the `port_` reference, offsetting and cast into a separate function. It cleaned up the code pretty substantially.

File: src/bno055_uart.cpp [hunk 6]
```diff
@@ -118,32 +128,40 @@ void BNO055UART::write(uint8_t page)
...
     std::stringstream ss;
-    ss << "Register write failed with error code: 0x" << std::hex << std::setw(2) << std::setfill('0') << write_response_.response_status; 
+    ss << "Register write failed with error code: 0x" << std::hex << std::setw(2) <<
+      std::setfill('0') << write_response_.response_status;
     throw serial::SerialException(ss.str().c_str());
   }
 }
 
+template<typename T>
+size_t BNO055UART::write(const T * data, const size_t offset, size_t size)
```
Comment (rotu) [inline src/bno055_uart.cpp:162]: nice! this is so much easier to reason about

File: LICENSE (hunk 1 above)
Comment (rotu) [reply LICENSE:179]: ~~I see why you did it that way, and if you feel strongly, I won't push the issue. And it seems ament_copyright passes even if the license file is missing or modified, or doesn't match the license declared in the `package.xml`.~~

**Review by rotu — APPROVED** [review]
~~I still don't agree about the license but~~ I approve these changes.
Weird - ament_copyright was being temporarily weird.

File: LICENSE (hunk 1 above)
Comment (rotu) [reply LICENSE:179]: Weird - I was wrong, and it seems ament_copyright was misbehaving for some reason or other.

**Pull request merged**